bot/cogs/news.py

```diff
@@ -0,0 +1,236 @@
+                    timestamp=new_date,
+                    url=link,
+                    author=f"{email_information['sender_name']} ({email_information['sender']['address']})",
+                    author_url=MAILMAN_PROFILE_URL.format(id=email_information["sender"]["mailman_id"]),
+                    webhook_profile_name=self.webhook_names[maillist],
+                    footer=f"Posted to {self.webhook_names[maillist]}"
+                )
+                payload["data"][maillist].append(thread_information["thread_id"])
+
+                if msg.channel.is_news():
+                    log.trace("Publishing mailing list message because it was in a news channel")
+                    await msg.publish()
+
+        await self.bot.api_client.put("bot/bot-settings/news", json=payload)
+
+    async def send_webhook(self,
+                           title: str,
+                           description: str,
+                           timestamp: datetime,
+                           url: str,
+                           webhook_profile_name: str,
+                           footer: str,
+                           author: t.Optional[str] = None,
+                           author_url: t.Optional[str] = None,
+                           ) -> discord.Message:
+        """Send webhook entry and return sent message."""
+        embed = discord.Embed(
+            title=title,
+            description=description,
+            timestamp=timestamp,
+            url=url,
+            colour=constants.Colours.soft_green
+        )
+        if author and author_url:
+            embed.set_author(
+                name=author,
+                url=author_url
+            )
+        embed.set_footer(text=footer, icon_url=AVATAR_URL)
+
+        return await self.webhook.send(
+            embed=embed,
+            username=webhook_profile_name,
+            avatar_url=AVATAR_URL,
+            wait=True
+        )
+
+    async def get_thread_and_first_mail(self, maillist: str, thread_identifier: str) -> t.Tuple[t.Any, t.Any]:
+        """Get mail thread and first mail from mail.python.org based on `maillist` and `thread_identifier`."""
+        async with self.bot.http_session.get(
+                THREAD_TEMPLATE_URL.format(name=maillist, id=thread_identifier)
+        ) as resp:
+            thread_information = await resp.json()
+
+        async with self.bot.http_session.get(thread_information["starting_email"]) as resp:
+            email_information = await resp.json()
+        return thread_information, email_information
+
+    async def get_webhook_and_channel(self) -> None:
+        """Storage #python-news channel Webhook and `TextChannel` to `News.webhook` and `channel`."""
+        await self.bot.wait_until_guild_available()
+        self.webhook = await self.bot.fetch_webhook(constants.PythonNews.webhook)
+        self.channel = await self.bot.fetch_channel(constants.PythonNews.channel)
```

Comment:
With the new `wait_until_guild_available` method we have in `Bot`, I don't think we need to fetch channels explicitly anymore. We can simply get it from the cache instead of making additional API calls. (Note: We should probably check the rest of our bot to see if there are more unnecessary `fetch_channel` calls.)